pkg/kafsar/group_coordinator_standalone.go

```diff
@@ -525,3 +573,59 @@ func (g *GroupCoordinatorStandalone) checkMemberExist(group *Group, memberId str
 	_, exist := group.members[memberId]
 	return exist
 }
+
+func (g *GroupCoordinatorStandalone) awaitingJoin(group *Group, memberId string, rebalanceTickMs int, sessionTimeout int) error {
+	start := time.Now()
+	for {
+		group.groupMemberLock.Lock()
+		if group.members[memberId].joinGenerationId != group.generationId {
+			group.members[memberId].joinGenerationId = group.generationId
+		}
```

Comment:
duplicate code in awaitingJoin and awaiting sync,

Reply:
resolve